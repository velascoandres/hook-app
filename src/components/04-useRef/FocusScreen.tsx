import React, { useRef } from 'react'
import { LegacyRef } from 'react';
import '../02-useEffect/effects.css';

export const FocusScreen = () => {

    // Referencias para componentes, numeros, etc
    const inputRef = useRef<HTMLInputElement>(null);


    const handleClick = () => {
        inputRef.current?.select();
    };


    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />

            <input
                ref={inputRef}
                type="text"
                className="form-control"
                placeholder="Su nombre"
            />

            <button
                className="btn btn-outline-primary mt-5"
                onClick={handleClick}
            >
                Focus
            </button>
        </div>
    );
}
