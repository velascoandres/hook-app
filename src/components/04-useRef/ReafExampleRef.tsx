import React, { useState } from 'react'
import '../02-useEffect/effects.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';


export const ReafExampleRef = () => {

    const [show, setshow] = useState<boolean>(true);

    return (
        <div>
            <h1>Real Example Ref</h1>
            <hr />

            { 
                show && 
                <MultipleCustomHooks />
            
            }


            <button
                className="btn btn-danger mt-5"
                onClick={() => setshow(!show)}
            >
                { show ? 'Hide' : 'Show'} 
            </button>

        </div>
    );
}
