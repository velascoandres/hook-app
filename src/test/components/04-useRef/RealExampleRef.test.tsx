import { shallow } from "enzyme";
import { ReafExampleRef } from "../../../components/04-useRef/ReafExampleRef";

describe('Pruebas en <RealExample />', () => {
    
    const wrapper = shallow(<ReafExampleRef />);

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });


    test('Debe ocultar el componente <MultipleCustomHokks />', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });

    
});
