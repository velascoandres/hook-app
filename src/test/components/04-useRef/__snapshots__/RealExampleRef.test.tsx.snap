// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExample /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-danger mt-5"
    onClick={[Function]}
  >
    Hide
  </button>
</div>
`;
